src/config.coffee

```diff
@@ -241,10 +241,13 @@ module.exports = class Config extends EventEmitter
 		deviceRegister.generateUniqueKey()
 
 	generateRequiredFields: =>
-		@getMany([ 'uuid', 'deviceApiKey', 'apiSecret', 'logsChannelSecret' ])
-		.then ({ uuid, deviceApiKey, apiSecret, logsChannelSecret }) =>
-			if !uuid? or !deviceApiKey? or !apiSecret? or !logsChannelSecret?
-				uuid ?= @newUniqueKey()
+		@getMany([ 'uuid', 'deviceApiKey', 'apiSecret', 'logsChannelSecret', 'offlineMode' ])
+		.then ({ uuid, deviceApiKey, apiSecret, logsChannelSecret, offlineMode }) =>
+			if !uuid?
+				uuid = @newUniqueKey()
+			return if offlineMode
+
+			if !deviceApiKey? or !apiSecret? or !logsChannelSecret?
 				deviceApiKey ?= @newUniqueKey()
 				apiSecret ?= @newUniqueKey()
```

Comment:
We also need the apiSecret, otherwise the supervisor API will break (and preloaded apps should be able to use the  supervisor API).